src/components/Reports/ReportsPage.vue

```diff
@@ -67,42 +72,109 @@
           />
         </el-form-item>
       </el-form>
-      <el-button
-        class="fullwidth black clickable"
-        style="margin-top: 40px;"
-      >
-        <fa-icon icon="download"/>
-        DOWNLOAD REPORT
-      </el-button>
+
+      <el-row :gutter="20">
+        <el-col :span="12">
+          <el-button
+            class="fullwidth black clickable"
+            style="margin-top: 40px;"
+            @click="download({ date, assetId: selectedWallet }, 'pdf')"
+          >
+            <fa-icon icon="download"/>
+            PDF
+          </el-button>
+        </el-col>
+
+        <el-col :span="12">
+          <el-button
+            class="fullwidth black clickable"
+            style="margin-top: 40px;"
+            @click="download({ date, assetId: selectedWallet }, 'csv')"
+          >
+            <fa-icon icon="download"/>
+            CSV
+          </el-button>
+        </el-col>
+      </el-row>
     </el-dialog>
   </div>
 </template>
 
 <script>
-import { mapGetters } from 'vuex'
+import { mapState, mapGetters } from 'vuex'
+import { generatePDF, generateCSV } from '@util/report-util'
+import dateFormat from '@/components/mixins/dateFormat'
+import FileSaver from 'file-saver'
+import { subMonths, startOfMonth, endOfMonth } from 'date-fns'
 
 export default {
   name: 'reports-page',
+  mixins: [dateFormat],
   data () {
     return {
       reportFormVisible: false,
-      selectedWallets: [],
+      selectedWallet: null,
       date: ''
     }
   },
   computed: {
-    ...mapGetters({
-      wallets: 'wallets'
+    ...mapState({
+      accountId: state => state.Account.accountId
     }),
-    mockReports: function () {
-      return this.wallets.map(x => ({
-        wallet: `${x.name} (${x.asset.toUpperCase()})`,
-        date: 'DEC 3, 2017 — JAN 3, 2018'
-      }))
+    ...mapGetters([
+      'wallets',
+      'settingsView',
+      'portfolioList'
+    ]),
+    previousMonthReports: function () {
+      return this.wallets.map(x => {
+        const previousMonth = subMonths(new Date(), 1)
+
+        return {
+          assetId: x.assetId,
+          walletName: `${x.name} (${x.asset.toUpperCase()})`,
+          date: [
+            startOfMonth(previousMonth),
+            endOfMonth(previousMonth)
+          ]
+        }
+      })
     }
   },
   created () {
     this.$store.dispatch('getAccountAssets')
+  },
+  methods: {
+    download ({ date, assetId }, fileFormat) {
+      Promise.all([
+        this.$store.dispatch('loadDashboard'),
+        this.$store.dispatch('getAccountAssetTransactions', { assetId })
+      ])
+        .then(() => {
+          const [dateFrom, dateTo] = date
+          const wallet = this.wallets.find(x => (x.assetId === assetId))
+          const priceFiat = this.portfolioList.find(({ asset }) => asset === wallet.asset).price
```

Comment:
(I'm writing it to let you know what I thought)

At first, I wondered whether it is ok to use `portfolioList` here because portfolioList, a getter of Dashboard store, is data for the portfolio as its name implies. In other words, Dashboard store was desgined in terms of view, not in terms of domain (of course, I don't say it is a problem at all) so I wondered if using it here really makes sense. However, the name aside, I think it seems ok since `portfolioList` includes sufficient data.

This is close to my consideration.
https://stackoverflow.com/questions/38208546/should-i-structure-the-redux-store-after-the-domain-or-after-the-app-views

Reply:
Yes, this is to ensure new prices to be fetched.

Reply:
Oh, I did a stupid thing. I found that it cannot ensure the order since `loadDashboard` doesn't return a Promise.